src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,745 @@
+    //    |XXXXXXXX|..<*..|    |XXXXXXXX|.<-*..|    |XXXXXXXX|<--*..|    |XXXXX|<-----*..|
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    +---------------+    +---------------+    +---------------+    +---------------+
+
+    const openValue = useDerivedValue(() => {
+      return interpolate(
+        drawerTranslation.value,
+        [0, drawerWidth],
+        [0, 1],
+        Extrapolation.CLAMP
+      );
+    });
+
+    const gestureOptions: {
+      useNativeDriver: boolean;
+      listener?: (event: { nativeEvent: { translationX: number } }) => void;
+    } = React.useMemo(
+      () => ({
+        useNativeDriver: useNativeAnimations,
+        listener: props.onDrawerSlide
+          ? (ev) => {
+              const translationX = Math.floor(
+                Math.abs(ev.nativeEvent.translationX)
+              );
+              const position = translationX / containerWidth;
+
+              props.onDrawerSlide?.(position);
+            }
+          : undefined,
+      }),
+      [containerWidth, props, useNativeAnimations]
+    );
+
+    if (props.onDrawerSlide) {
+      gestureOptions.listener = (ev) => {
+        const translationX = Math.floor(Math.abs(ev.nativeEvent.translationX));
+        const position = translationX / containerWidth;
+
+        props.onDrawerSlide?.(position);
+      };
+    }
+
+    const isDrawerOpen = useSharedValue(false);
+
+    const handleContainerLayout = ({ nativeEvent }: LayoutChangeEvent) => {
+      setContainerWidth(nativeEvent.layout.width);
+    };
+
+    const emitStateChanged = React.useCallback(
+      (newState: DrawerState, drawerWillShow: boolean) => {
+        'worklet';
+        props.onDrawerStateChanged &&
+          runOnJS(props.onDrawerStateChanged)?.(newState, drawerWillShow);
+      },
+      [props.onDrawerStateChanged]
+    );
+
+    const isDrawerShowing = useSharedValue(false);
+
+    const drawerAnimatedProps = useAnimatedProps(() => ({
+      accessibilityViewIsModal: isDrawerShowing.value,
+    }));
+
+    const overlayAnimatedProps = useAnimatedProps(() => ({
+      pointerEvents: isDrawerShowing.value
+        ? ('auto' as const)
+        : ('none' as const),
+    }));
+
+    // While the drawer is hidden, it's hitSlop overflows onto the main view by edgeWidth
+    // This way it can be swiped open even when it's hidden
+    const [edgeHitSlop, setEdgeHitSlop] = React.useState<HitSlop>(
+      isFromLeft ? { right: edgeWidth } : { left: edgeWidth }
+    );
+
+    // gestureOrientation is 1 if the gesture is expected to move from left to right and -1 otherwise
+    const gestureOrientation = React.useMemo(
+      () => (isFromLeft ? 1 : -1) * (drawerOpened ? -1 : 1),
+      [isFromLeft, drawerOpened]
+    );
+
+    const animateDrawer = React.useCallback(
+      (toValue: number, velocity: number, speed?: number) => {
```

Comment:
What's the difference between `speed` and `velocity`?

Reply:
`velocity` is the initial animation velocity, `speed` is the animation speed.
This is an artifact from the legacy component, i can rename it internally to `animationSpeed`, 
but since it's already exposed in the `DrawerLayoutProps` interface, i'm not sure if we should change it there.

Reply:
I'd go further and also change `velocity` to `initialVelocity`. The way it's been in the previous implementation is really confusing.

Reply:
Changed in [82e3c00](https://github.com/software-mansion/react-native-gesture-handler/pull/3146/commits/82e3c00bc2b66eb576ed23757ccb91f650b102cf)

The exposed `type` still exposes `speed` and `velocity`, let me know if we want to deprecate the exported `velocity` and `speed` props as well, and alias them with `initialVelocity` and `animationSpeed`.

Reply:
Why not just remove them and use the new ones? Since this is a new component it wouldn't be a breaking change.

Reply:
Alright, just to clarify the component is not entirely new - it's a replacement for existing `DrawerLayout`.
This particular type has been pulled directly from the legacy version. 
Regardless, I'm all for it, the change is minimal, and if anything breaks for anyone, the fix is trivial.

Changed in [5b0237e](https://github.com/software-mansion/react-native-gesture-handler/pull/3146/commits/5b0237e4f843df8d88a2b7d7fb8eaf01f91acd1f)